### PR TITLE
Check for static imports in unused imports rule

### PR DESCRIPTION
**Pull request by sowmyav24**
Fixes [#3131 ](https://github.com/detekt/detekt/issues/3131)

### CODE REVIEW

**Review by BraisGabin — APPROVED** [review]

**Comment by sowmyav24** [comment]
When I run the task `./gradlew :detekt-rules-style:test` in local it passes. In CI it seems to fails for test-snippet compilation.

**Comment by BraisGabin** [comment]
Use `./gradlew :detekt-rules-style:test -Pcompile-test-snippets=true`

Compile all the snippets each time that we run the tests slow down the test and is kind of meta testing (testing the tests). For that reason we have the compilation disabled by default and we just compile them in CI.

`import org.eclipse.persistence.annotations.FetchType` is not in the classpath so it fails. `compileAndLintWithContext` allows multiple `Strings` that are like different files. This way you can create your own enum in other package and then use it in your test.

**Review by cortinico — APPROVED** [review]

**Review by schalkms — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt [hunk 1]
```diff
@@ -552,5 +552,21 @@ class UnusedImportsSpec : Spek({
...
+                import x.y.z.Foo
+                
+                val x = Foo.LAZY
+            """
+            val additionalFile = """
+                package x.y.z
+                
+                enum class FetchType {
+                    LAZY
+                }
```
Comment (BraisGabin) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt:567]: Why does this compile? the enum is called `FetchType` but the import is for `Foo` 🤔

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt (hunk 1 above)
Comment (sowmyav24) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt:567]: I wrote the tests with Foo and later renamed them before committing  to mimic the issue reported. However, your point of compilation seems to be valid 😮

Comment (BraisGabin) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt]: @schalkms do you have any idea about why does this happen? This doesn't stop the PR but I will like to know why does this happen

Comment (schalkms) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedImportsSpec.kt]: @BraisGabin Yes, I do. Please take a look at the following lines and closely check line number 68.

https://github.com/detekt/detekt/blob/da6259a15c986d248799cc5f2db308689e4fc68e/detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/RuleExtensions.kt#L63-L72

**Review by arturbosch — APPROVED** [review]

**Pull request merged**